src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -209,48 +211,201 @@ internal static IEnumerable<string> GetCommaSeparatedValues(string text)
             var accessibleProperties = type.GetRuntimeProperties().Where(IsAccessibleProperty);
             var attributeDescriptors = new List<TagHelperAttributeDescriptor>();
 
+            // Keep indexer descriptors separate to avoid sorting the combined list later.
```

Comment:
Not sure what you mean. This is about having `asp-route-name` matching its own property even if a dictionary property with prefix `asp-route-` exists. And the semantic I've implemented here is exactly what we decided on.